modules/ROOT/pages/salesforce/salesforce-connector-96.adoc

```diff
@@ -0,0 +1,1211 @@
+https://www.mulesoft.com/exchange/org.mule.examples/import-leads-into-salesforce/[import leads]
+* https://www.mulesoft.com/exchange/org.mule.examples/cache-scope-with-salesforce-contacts/[Contacts]
+* https://www.mulesoft.com/exchange/org.mule.examples/salesforce-data-retrieval/[Data retrieval]
+* https://www.mulesoft.com/exchange/org.mule.examples/dataweave-with-flowreflookup/[DataWeave]
+
+== Develop an Application
+
+Build an application in this order:
+
+. Configure the connector.
+. Test the connection.
+. Build the rest of your flow.
+. Add and configure DataWeave.
+
+When developing an app, the Salesforce connector provides these integration patterns
+you can use with the Salesforce APIs:
+
+* Batch Data Synchronization - An external system accesses, changes, deletes, or adds data in Salesforce in batches, and vice versa (Salesforce to external system).
+* Remote Call-In - An external system accesses, changes, deletes or adds data in Salesforce, and vice versa (Salesforce to external system).
+* Fire and Forget Remote Process Invocation - Salesforce initiates a process in a third-party system and receives an acknowledgment that the process has started. The third-party system continues processing independent of Salesforce.
+* Request-Reply Remote Process Invocation - Salesforce kicks off a process in a remote system, waits for the remote system to finish processing, then accepts control back again from the remote system.
+* User Interface Update Based on Data Changes - The Salesforce UI updates in response to a change in a third-party system.
+
+The Salesforce connector works with these APIs:
+
+*Apex SOAP API:*
+
+The Salesforce https://www.salesforce.com/us/developer/docs/apexcode/[Apex SOAP API] exposes Apex class methods as custom SOAP Web service calls. This allows an external application to invoke an Apex Web service to perform an action in Salesforce.
+
+*SOAP API:*
+
+The Salesforce http://www.salesforce.com/us/developer/docs/api/index.htm[SOAP API] provides secure access to your organization's information on Salesforce. Most of the operations that the Salesforce connector performs map to this API.
+
+All the Salesforce operations performed through the SOAP API have an optional parameter called "Headers" that can take any of the following https://developer.salesforce.com/docs/atlas.en-us.api.meta/api/soap_headers.htm[Salesforce SOAP Headers]:
+AllOrNoneHeader, AllowFieldTruncationHeader, AssignmentRuleHeader, CallOptions,
+EmailHeader, LocaleOptions, MruHeader, OwnerChangeOptions, QueryOptions,
+UserTerritoryDeleteHeader, DuplicateRuleHeader
+
+*Apex REST API:*
+
+The Salesforce https://developer.salesforce.com/page/Creating_REST_APIs_using_Apex_REST[Apex REST API] creates your own REST-based web services using Apex. It has all of the advantages of the REST architecture, and provides the ability to define custom logic and includes automatic argument/object mapping.
```

Comment:
It seems strange that the Apex SOAP API and Apex REST API intro sentences aren't more parallel.